pkg/uniqueid/instance_registerer.go

```diff
@@ -0,0 +1,96 @@
+package uniqueid
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/applike/gosoline/pkg/cfg"
+	"github.com/applike/gosoline/pkg/clock"
+	"github.com/applike/gosoline/pkg/cloud/aws/ecs"
+	"github.com/applike/gosoline/pkg/log"
+)
+
+const ErrorMaxInstancesRegistered = errorMaxInstancesRegistered("max instances registered")
+
+type errorMaxInstancesRegistered string
+
+func (e errorMaxInstancesRegistered) Error() string {
+	return string(e)
+}
+
+//go:generate mockery --name InstanceRegisterer
+type InstanceRegisterer interface {
+	Register(ctx context.Context) (*Instance, error)
+	Unregister(ctx context.Context, instance Instance) error
+}
+
+type instanceRegisterer struct {
+	clock      clock.Clock
+	repository InstanceRepository
+	settings   Settings
+}
+
+func NewInstanceRegisterer(ctx context.Context, config cfg.Config, logger log.Logger, settings Settings) (InstanceRegisterer, error) {
+	clk := clock.NewRealClock()
+
+	repository, err := NewInstanceRepository(ctx, config, logger)
+	if err != nil {
+		return nil, fmt.Errorf("could not create repository: %w", err)
+	}
+
+	if settings.TaskARN == "" {
+		ecsMetadata, err := ecs.ReadEcsMetadata()
+		if err != nil {
+			return nil, fmt.Errorf("could not read ecs metadata: %w", err)
+		}
+
+		taskArn, ok := ecsMetadata["TaskARN"]
+		_, isStr := taskArn.(string)
+		if !ok || !isStr {
+			return nil, fmt.Errorf("invalid task arn")
+		}
+
+		settings.TaskARN = taskArn.(string)
+	}
+
+	return NewInstanceRegisterWithInterfaces(clk, repository, settings), nil
+}
+
+func NewInstanceRegisterWithInterfaces(clk clock.Clock, repository InstanceRepository, settings Settings) *instanceRegisterer {
+	return &instanceRegisterer{
+		clock:      clk,
+		repository: repository,
+		settings:   settings,
+	}
+}
+
+func (r *instanceRegisterer) Register(ctx context.Context) (*Instance, error) {
+	for i := uint16(0); i < r.settings.MaxInstances; i++ {
```

Comment:
So how does this work in detail if I have a service with 500 instances running? Will this mean I have to make 500 calls?